consensus/state_test.go

```diff
@@ -2659,6 +2656,21 @@ func subscribeUnBuffered(eventBus *types.EventBus, q cmtpubsub.Query) <-chan cmt
 	return sub.Out()
 }
 
+func waitForProposal(t *testing.T, proposalCh <-chan cmtpubsub.Message, height int64, round int32) []byte {
+	t.Helper()
+	select {
+	case <-time.After(ensureTimeout):
+		t.Fatal("timeout expired while waiting for proposal event")
+	case msg := <-proposalCh:
+		proposalEvent, ok := msg.Data().(types.EventDataCompleteProposal)
+		require.True(t, ok, "expected EventDataCompleteProposal, got %T", msg.Data())
+		require.Equal(t, height, proposalEvent.Height, "proposal height mismatch")
+		require.Equal(t, round, proposalEvent.Round, "proposal round mismatch")
+		return proposalEvent.BlockID.Hash
+	}
+	return nil
+}
```

Comment:
the thinking here is that there is a the blockID isn't updated or is updated too quickly, especially since we're using the prop reactor and funneling the proposals through an extra channel.

locally its always fast enough to never occur, but it in CI with millicpu from kupernetes we get rekt